regnety/dataset/imagenet.py

```diff
@@ -0,0 +1,268 @@
+                    )
+                if w_cropped == w:
+                    x = tf.constant(0, dtype=tf.int64)
+                else:
+                    x = tf.random.uniform(
+                        (), minval=0, maxval=w - w_cropped, dtype=tf.int64
+                    )
+
+                image = image[y : (y + h_cropped), x : x + w_cropped, :]
+                image = tf.image.resize(
+                    image, (self.image_size, self.image_size)
+                )
+                break
+        if return_example:
+            return {
+                "image": image,
+                "height": self.image_size,
+                "width": self.image_size,
+                "filename": example["filename"],
+                "label": example["label"],
+                "synset": example["synset"],
+            }
+
+        image = self._scale_and_center_crop(
+            image, h, w, self.image_size, self.image_size
+        )
+        return {
+            "image": image,
+            "height": self.image_size,
+            "width": self.image_size,
+            "filename": example["filename"],
+            "label": example["label"],
+            "synset": example["synset"],
+        }
+
+    def clip_example(self, example):
+        """Takes an example having keys 'image' and 'label' and returns example
+        with keys 'image' and 'target'. 'target' is one hot encoded.
+
+        Args:
+            example: an example having keys 'image' and 'label'
+
+        Returns:
+            example having keys 'image' and 'target'
+        """
+        return {
+            "image": example["image"],
+            "target": tf.one_hot(example["label"], self.num_classes),
+        }
+
+    def _randaugment(self, example):
+        """Wrapper for tf vision's RandAugment.distort function which
+        accepts examples as input instead of images. Uses magnitude = 5
+        as per pycls/pycls/datasets/augment.py#L29.
+
+        Args:
+            example: Example having the key 'image'
+
+        Returns:
+            example in which RandAugment has been applied to the image
+        """
+        example['image'] = self.augmenter.distort(example['image'])
+        return example
+
+    def make_dataset(
+        self,
+    ):
+        """
+        Function to apply all preprocessing and augmentations on dataset using
+        dataset.map().
+
+        Returns:
+            Dataset having the final format as follows:
+            {
+                'image' : (self.image_size, self.image_size, 3)
+                'target' : (num_classes,)
+            }
+        """
+        ds = self._read_tfrecs()
+
+        if self.augment_fn == "default":
+            ds = ds.map(lambda example: self.random_sized_crop(example))
+            if self.randaugment:
+                ds = ds.map(lambda example: self._randaugment(example))
+            #ds = ds.map(lambda example: self.clip_example(example))
```

Comment:
Should this be deleted? Also consider deleting the `self.clip_example` method if it's unused